.github/workflows/workspace_publish.yml

```diff
@@ -37,9 +37,9 @@ jobs:
         run: deno test --no-run --doc
 
       - name: Publish (dry run)
-        if: startsWith(github.ref, 'refs/tags/') == false
+        if: github.event_name == 'push'
         run: deno publish --dry-run
 
       - name: Publish (real)
-        if: startsWith(github.ref, 'refs/tags/')
+        if: github.event_name == 'release'
```

Comment:
You can find the example of this expression, for example, in https://github.com/google/brotli/blob/ccec9628e49208173c9a8829ff66d0d7f6ceff5f/.github/workflows/release.yaml#L118